nucleus/payara-modules/healthcheck-core/src/main/java/fish/payara/nucleus/healthcheck/preliminary/MachineMemoryUsageHealthCheck.java

```diff
@@ -91,110 +91,111 @@ public String getDescription() {
+		            }
+		            if (INACTIVEFILE.equals(part)) {
+		                memInactiveFile = parseMemInfo(parts);
+		            }
+		            if (RECLAIMABLE.equals(part)) {
+		                memReclaimable = parseMemInfo(parts);
+		            }
+		        }
+		    }
+
+		    if (!memAvailableFound) {
+		        memAvailable = memFree + memActiveFile + memInactiveFile + memReclaimable;
+		    }
+
+		    double usedPercentage = memTotal == 0L ? 0d :((double) (memTotal - memAvailable) / memTotal) * 100;
+
+		    result.add(new HealthCheckResultEntry(decideOnStatusWithRatio(usedPercentage),
+		            "Physical Memory Used: " + prettyPrintBytes(memTotal - memAvailable) + " - " +
+		                    "Total Physical Memory: " + prettyPrintBytes(memTotal) + " - " +
+		                    "Memory Used%: " + new DecimalFormat("#.00").format(usedPercentage) + "%"));
+
+		} catch (IOException exception) {
+		    result.add(new HealthCheckResultEntry(HealthCheckResultStatus.CHECK_ERROR,
+		            "Memory information cannot be read for retrieving physical memory usage values",
+		            exception));
+		} catch (ArithmeticException exception) {
+		    result.add(new HealthCheckResultEntry(HealthCheckResultStatus.CHECK_ERROR,
+		            "Error occurred while calculating memory usage values. Total memory is " + memTotal,
+		            exception));
+		}
+		return result;
+	}
+
+	private HealthCheckResult doCheckNonLinux() {
+		HealthCheckResult result = new HealthCheckResult();
+		try {
+		    OperatingSystemMXBean osBean = ManagementFactory.getOperatingSystemMXBean();
+		    Long totalPhysicalMemSize = invokeMethodFor(osBean, "getTotalPhysicalMemorySize");
+		    Long freePhysicalMemSize = invokeMethodFor(osBean, "getFreePhysicalMemorySize");
+
+		    double usedPercentage = ((double) (totalPhysicalMemSize - freePhysicalMemSize) / totalPhysicalMemSize) *
+		            100;
+
+		    result.add(new HealthCheckResultEntry(decideOnStatusWithRatio(usedPercentage),
+		            "Physical Memory Used: " + prettyPrintBytes((totalPhysicalMemSize - freePhysicalMemSize)) + " - " +
+		                    "Total Physical Memory: " + prettyPrintBytes(totalPhysicalMemSize) + " - " +
+		                    "Memory Used%: " + new DecimalFormat("#.00").format(usedPercentage) + "%"));
+
+		} catch (Exception exception) {
+		    result.add(new HealthCheckResultEntry(HealthCheckResultStatus.CHECK_ERROR,
+		            "Operating system methods cannot be invoked for retrieving physical memory usage values",
+		            exception));
+		}
+		return result;
+	}
+
+    private static boolean isLinux() {
         String osName = System.getProperty("os.name");
-        if (osName.startsWith("Linux") ||
+        return osName.startsWith("Linux") ||
                 osName.startsWith("FreeBSD") ||
                 osName.startsWith("OpenBSD") ||
                 osName.startsWith("gnu") ||
                 osName.startsWith("gnu/kfreebsd") ||
```

Comment:
Isn't this line redundant?

Reply:
You are right. It should be ok to remove it without change of behaviour.

Reply:
Removed the `gnu/kfreebsd` as it is included in `gnu`.